src/components/cards/challenge/_partials/Button.tsx

```diff
@@ -2,23 +2,32 @@ import { useState } from "react";
 import CloseIcon from "@/icons/close-top-right.svg";
 import Loader from "@/components/ui/Loader";
 
-interface ButtonProps {
+export interface ButtonProps {
   text: string;
   onClick: () => void;
   loading: boolean;
+  buttonTestId?: string;
 }
 
-export default function Button({ text, onClick, loading }: ButtonProps) {
-  const [isTextVisible, setistextVisible] = useState(false);
+export default function Button({ text, onClick, loading, buttonTestId }: ButtonProps) {
+  const [isTextVisible, setIsTextVisible] = useState(false);
```

Comment:
this prop should be `testId`. Please fix the linting and prettier issues in this component